docs/features/deconstruction.md

```diff
@@ -121,61 +121,63 @@ Note that tuples (`System.ValueTuple`) don't need to invoke Deconstruct.
 declaration_statement
     : local_variable_declaration ';'
     | local_constant_declaration ';'
-    | local_variable_combo_declaration ';'  // new
     ;
 
-local_variable_combo_declaration
-    : local_variable_combo_declaration_lhs '=' expression
+local_variable_declaration
+	: local_variable_type local_variable_declarators
+	| deconstruction_declaration // new
+	;
 
-local_variable_combo_declaration_lhs
-    : 'var' '(' identifier_list ')'
-    | '(' local_variable_list ')'
-    ;
+deconstruction_declaration // new
+	: deconstruction_variables '=' expression
+	;
 
-identifier_list
-    : identifier ',' identifier
-    | identifier_list ',' identifier
-    ;
+deconstuction_variables
+	: '(' deconstuction_variables_nested (',' deconstuction_variables_nested)* ')'
+	| 'var' deconstruction_identifiers
+	;
 
-local_variable_list
-    : local_variable_type identifier ',' local_variable_type identifier
-    | local_variable_list ',' local_variable_type identifier
-    ;
+deconstuction_variables_nested // new
+	: deconstuction_variables
+	| type identifier
+	;
+
+deconstruction_identifiers
+	: '(' deconstruction_identifiers_nested (',' deconstruction_identifiers_nested)* ')'
+	;
+
+deconstruction_identifiers_nested // new
+	: deconstruction_identifiers
+	| identifier
+	;
 
 foreach_statement
     : 'foreach' '(' local_variable_type identifier 'in' expression ')' embedded_statement
-    | 'foreach' '(' local_variable_combo_declaration_lhs 'in' expression ')' embedded_statement // new
+    | 'foreach' '(' deconstruction_variables 'in' expression ')' embedded_statement // new
+    ;
+
+for_statement
+    : 'for' '(' for_initializer? ';' for_condition? ';' for_iterator? ')' embedded_statement
     ;
```

Comment:
In the grammar, I kept `for_statement` as before. The difference appears below, in `for_initializer`.